lib/phoenix/debug.ex

```diff
@@ -57,32 +96,74 @@ defmodule Phoenix.Debug do
   @doc """
   Returns a list of all currently connected channels for the given `Phoenix.Socket` pid.
 
+  Each channel is represented as a map with the following keys:
+
+    - `:pid` - the pid of the channel process
+    - `:status` - the status of the channel
+    - `:topic` - the topic of the channel
+    - `:type` - the type of the channel.
+      It is `:channel` for channels defined with `use Phoenix.Channel` and `:custom` for custom channels (like LiveViews)
+
   ## Examples
 
-      iex> Phoenix.Debug.list_channel_sockets() |> Enum.at(0) |> Phoenix.Debug.channels()
+      iex> pid = Phoenix.Debug.list_channel_sockets() |> Enum.at(0) |> Map.fetch!(:pid)
+      iex> Phoenix.Debug.list_channels(pid)
       {:ok,
        [
-         %{pid: #PID<0.1702.0>, status: :joined, topic: "lv:phx-GDp9a9UZPiTxcgnE"},
-         %{pid: #PID<0.1727.0>, status: :joined, topic: "lv:sidebar"}
+         %{pid: #PID<0.1702.0>, status: :joined, topic: "lv:phx-GDp9a9UZPiTxcgnE", type: :custom},
+         %{pid: #PID<0.1727.0>, status: :joined, topic: "lv:sidebar", type: :custom}
        ]}
 
-      iex> Phoenix.Debug.channels(pid(0,456,0))
+      iex> Phoenix.Debug.list_channels(pid(0,456,0))
       {:error, :not_alive}
 
   """
-  def channels(socket_pid) do
+  def list_channels(socket_pid) do
     ref = make_ref()
 
-    if Process.alive?(socket_pid) do
+    if Process.alive?(socket_pid) and channel_socket?(socket_pid) do
       send(socket_pid, {:debug_channels, ref, self()})
 
       receive do
-        {:debug_channels, ^ref, channels} -> {:ok, channels}
+        {:debug_channels, ^ref, channels} ->
+          channels =
+            Enum.map(channels, fn %{pid: pid} = info ->
+              Map.put(info, :type, (channel?(pid) && :channel) || :custom)
+            end)
```

Comment:
Perhaps we should not do this additional work by default. Instead, return as is and we document that they can use `channel?` to check.